File: apple/Sources/ReactNativeWatchOS/Modifiers/ContainerValue.swift
import SwiftUI

/// SwiftUI `.containerValue(_:_:)` (watchOS 11+).
///
/// LIMITATION — implemented as a no-op. `containerValue` writes into a
/// `ContainerValues` slot addressed by a static `WritableKeyPath` declared
/// in Swift (via `@Entry`). There is no general, type-safe way to resolve
/// such a key path from a runtime string arriving across the JS bridge, and
/// values custom containers read are bridge-opaque. The registration exists
/// so the `$type` is recognized (forward-compat) and the view passes through
/// unchanged rather than being dropped.
enum RNWContainerValueModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("containerValue") { view, _, _ in
            // No generic bridge for ContainerValues key paths — see note above.
            return view
        }
    }
}

File: src/modifiers/containerValue.ts
import { createModifier } from './createModifier';

export interface ContainerValueParams {
  /// The container-values key to set. Only keys the native side knows how
  /// to resolve take effect.
  key: string;
  /// The value to associate with `key`.
  value?: unknown;
}

/// SwiftUI `.containerValue(_:_:)` (watchOS 11+). This sets a value for a
/// `ContainerValues` key path so a custom container can read it back.
///
/// LIMITATION: `ContainerValues` keys are defined by a Swift
/// `@Entry`/`EnvironmentKey`-style declaration and addressed by static
/// key path — there is no general way to look one up from a runtime string
/// across the bridge. The native side therefore implements this as a no-op
/// (the view is returned unchanged). Kept for API completeness so JS that
/// references it still type-checks.
export function containerValue(params: ContainerValueParams) {
  return createModifier('containerValue', params);
}
